packages/hackmode/lib/index.js

```diff
@@ -167,22 +158,26 @@ class N3hHackMode extends AsyncClass {
           })
           return
         case 'trackDna':
+          // Note: opt.data is a TrackDnaData
           this._track(opt.data.dnaAddress, opt.data.agentId)
           return
+        case 'untrackDna':
+          // Note: opt.data is a TrackDnaData
+          tId = this._getTransportIdOrFail(opt.data.dnaAddress, opt.data.agentId)
+          if (tId === null) {
+            return
+          }
+          this._untrack(opt.data.dnaAddress, opt.data.agentId)
+          return
         case 'sendMessage':
-          ref = this._getMemRef(opt.data.dnaAddress)
-          if (!(opt.data.toAgentId in ref.agentToTransportId)) {
-            log.w('NO ROUTE FOR sendMessage', opt.data)
-            this._ipc.send('json', {
-              method: 'failureResult',
-              dnaAddress: opt.data.dnaAddress,
-              _id: opt.data._id,
-              toAgentId: opt.data.fromAgentId,
-              errorInfo: 'No routing for agent id "' + opt.data.toAgentId + '" aborting sendMessage'
-            })
+          // Note: opt.data is a MessageData
+          if (this._getTransportIdOrFail(opt.data.dnaAddress, opt.data.fromAgentId, opt.data.fromAgentId, opt.data._id) === null) {
+            return
+          }
+          tId = this._getTransportIdOrFail(opt.data.dnaAddress, opt.data.toAgentId, opt.data.fromAgentId, opt.data._id)
+          if (tId === null) {
```

Comment:
Think some comments in here would help... I was very confused until I realized the `to` and `from` ids were swapped in these two calls

Reply:
Indeed it can be a bit confusing.
Fixed.